app/src/main/java/fr/free/nrw/commons/contributions/ContributionsPresenter.java

```diff
@@ -3,104 +3,156 @@
 
     private final ContributionsRepository repository;
+    private final Scheduler mainThreadScheduler;
+    private final Scheduler ioThreadScheduler;
+    private CompositeDisposable compositeDisposable;
     private ContributionsContract.View view;
-    private Cursor cursor;
+    private List<Contribution> contributionList=new ArrayList<>();
 
     @Inject
     Context context;
 
     @Inject
-    ContributionsPresenter(ContributionsRepository repository) {
+    UserClient userClient;
+
+    @Inject
+    AppDatabase appDatabase;
+
+    @Inject
+    SessionManager sessionManager;
+    private LifecycleOwner lifeCycleOwner;
+
+    @Inject
+    ContributionsPresenter(ContributionsRepository repository, @Named(CommonsApplicationModule.MAIN_THREAD) Scheduler mainThreadScheduler,@Named(CommonsApplicationModule.IO_THREAD) Scheduler ioThreadScheduler) {
         this.repository = repository;
+        this.mainThreadScheduler=mainThreadScheduler;
+        this.ioThreadScheduler=ioThreadScheduler;
     }
 
+    private String user;
+
     @Override
     public void onAttachView(ContributionsContract.View view) {
         this.view = view;
-        if (null != cursor) {
-            try {
-                cursor.registerDataSetObserver(this);
-            } catch (IllegalStateException e) {//Cursor might be already registered
-                Timber.d(e);
-            }
-        }
+        compositeDisposable=new CompositeDisposable();
     }
 
-    @Override
-    public void onDetachView() {
-        this.view = null;
-        if (null != cursor) {
-            try {
-                cursor.unregisterDataSetObserver(this);
-            } catch (Exception e) {//Cursor might not be already registered
-                Timber.d(e);
-            }
-        }
+    public void setLifeCycleOwner(LifecycleOwner lifeCycleOwner){
+        this.lifeCycleOwner=lifeCycleOwner;
     }
 
-    @NonNull
-    @Override
-    public Loader<Cursor> onCreateLoader(int id, @Nullable Bundle args) {
-        int preferredNumberOfUploads = repository.get(UPLOADS_SHOWING);
-        return new CursorLoader(context, BASE_URI,
-                ALL_FIELDS, "", null,
-                ContributionDao.CONTRIBUTION_SORT + "LIMIT "
-                        + (preferredNumberOfUploads>0?preferredNumberOfUploads:100));
+    public void fetchContributions() {
+        Timber.d("fetch  Contributions");
+        LiveData<List<Contribution>> liveDataContributions = repository.fetchContributions();
+        if(null!=lifeCycleOwner) {
+             liveDataContributions.observe(lifeCycleOwner, this::showContributions);
+        }
+
+        if (NetworkUtils.isInternetConnectionEstablished(CommonsApplication.getInstance()) && shouldFetchContributions()) {
+            Timber.d("fetching contributions: ");
+            view.showProgress(true);
+            this.user = sessionManager.getUserName();
+            view.showContributions(Collections.emptyList());
```

Comment:
you could just delete all entries in the DB here, that is functionally what you want to happen right? And then the livedata will emit the empty list for you. Hmmm no we are banking too much on that web request being successful.

Actually the more I think about this the weirder it is, "show nothing because we are loading" was this how it worked previously?

Reply:
The previous fetches were from a content sync-adapter but yes, the methods intentions were the same as the one we have now